float64 min_distance
string direction